.github/workflows/ad-hoc-deploy-preview.yml

```
@@ -12,38 +12,27 @@ on:
         description: "Flag to skip Cypress tests"
         required: true
         default: "true"
-      uid:
-        description: "Unique ID to store the run data in the db"
-        required: false
+      id:
+        description: "Document ID for the workflow entry in the MongoDB ( This gets initialized by the ops api )"
+        required: true
       sub-domain-name:
         description: "Sub-domain for dp.appsmith.com to by used by this deploy preview (This will also be the image name and the k8s namespace identifier)"
         required: true
 jobs:
-  # write-job-details-to-db:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - name: Install mongosh
-  #       run: |
-  #         sudo apt-get update
-  #         sudo apt-get install -y wget gnupg
-  #         wget -qO - https://www.mongodb.org/static/pgp/server-5.0.asc | sudo apt-key add -
-  #         echo "deb [ arch=amd64,arm64 ] https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/5.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-6.0.list
-  #         sudo apt-get update
-  #         sudo apt-get install -y mongodb-mongosh
-  #     - name: Update job data to mongoDB
-  #       run: |
-  #         #  mongosh '' --eval 'db.appsmiht-ce-github-dp-runs.insert( { run_id: ${{ github.run_id }}, uid: ${{ github.event.inputs.uid }}, status: "Started" } )'
-  #         mongosh '' --eval 'db.appsmiht-ce-github-dp-runs.update(
-  #                                                                 { _id:  ${{ github.event.inputs.uid }} },
-  #                                                                 {
-  #                                                                   $set: {
-  #                                                                     run_id: ${{ github.run_id }},
-  #                                                                     status: "BUILD IN PROGRESS"
-  #                                                                   }
-  #                                                                 }
-  #                                                               )'
-  #         #TODO Add mongo URI as secret
-
+  update-job-details-to-db:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update workflow details
+        run: |
+          curl --location --request PATCH 'https://ops.appsmith.com/appsmith_deploy_workflow/${{github.event.inputs.uid}}' \
+          --header 'Content-Type: application/json' \
+          --header 'Authorization: Bearer ${{secrets.INTENRAL_OPS_API_TOKEN}}' \
+          --data ' {
+                  "status": "IN_PROGRESS",
+                  "githubWorkflowId": ${{github.workflow.id}},
+                  "githubWorkflowHtmlUrl": ${{github.workflow.githubWorkflowHtmlUrl}},
```

Won't this be unquoted, and so result in invalid JSON?

```suggestion
                  "githubWorkflowHtmlUrl": "${{github.workflow.githubWorkflowHtmlUrl}}",
```

---

Noted.